### ISSUE TITLE
"Copy Code to Clipboard" Not Available at Start

### ISSUE THREAD

**Issue by aenuros** [facebook/Docusaurus#1039]
## 📚 Documentation

I installed Docusaurus per the instructions on the [installation page](https://docusaurus.io/docs/en/installation). I noticed that all the code blocks on the Docusaurus site and the example sites had a button on the upper right corner that you could click on to copy the code. However, when I created docs in markdown using the three backticks, the "Copy" button did not appear on any of them.

I went through the site to find out how to get the "Copy" button to show up. The only hint I could find was that Prism had a copy-to-clipboard.js in the node_module, and [this page](https://docusaurus.io/docs/en/site-config) said I could set Prism highlighting to true. It did indeed activate the Prism highlighting, but not the copy functionality I was looking for.

After searching all over the place, I found that I was missing some files by comparing my document to the code here. I needed to add this to siteConfig.js:

```
scripts: ['https://buttons.github.io/buttons.js',
  'https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.0/clipboard.min.js',
'/js/code-blocks-buttons.js'],
```
But I also needed to add [code-blocks-buttons.js](https://github.com/facebook/Docusaurus/blob/master/v1/website/static/js/code-blocks-buttons.js) to static/js, and [code-blocks-buttons.css](https://github.com/facebook/Docusaurus/blob/master/v1/website/static/css/code-blocks-buttons.css) to static/css. These files did not come with my installation of Docusaurus, nor did I find any reference on either the site nor Stack Overflow that I needed these files to have the copy code button working.

A guide on how to add the "Copy" button would be very useful. The fact that the Docusaurus site had the buttons was one of the things that led me to choose it over similar documentation options, but it was far from obvious to figure out what I needed to do to get the button to show up. I would volunteer to write the guide.

Alternatively, perhaps there was an issue with my installation and I was supposed to have these files and did not? Or is the "Copy" functionality is not available by default?

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes, I have read and understood the section pertaining to new issues.

**Comment by woollybah** [comment]
I also saw the Copy buttons and thought it would be a nice feature to add to my site too.
I ended up trawling through the Docusaurus sources to find all the bits I needed. Some documentation or a configuration option to enable it would be useful, I think.

It may also be worth adding something like the following to `code-blocks-buttons.css`, to remove the unsightly blue border that appears around the "button" when you click on it :-)
```
pre .btnIcon:focus {
    outline: 0;
}
```

**Comment by aenuros** [comment]
I've whipped up a [very quick guide](https://gist.github.com/aenuros/443a6f50cd72e5b7eab29331dee03219) to adding the "Copy Code to Clipboard" button in. I am not sure where in the Guide organization it would belong. I was thinking Guide > Custom Pages, but I don't know if that's the best location for it in the guide.

Perhaps all this will come by default in [Docusaurus V2](https://docusaurus.io/blog/2018/09/11/Towards-Docusaurus-2), but until then having something like this in the guide would be pretty useful.

**Comment by yangshun** [comment]
Closed by #1047
